File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int size = nums.size();
        vector<int> count(size+1 , 0);

        for(int i = 0 ; i<size ; i++){
            count[nums[i]]++;
        }

        for(int i = 0 ; i<size ; i++){
            if(count[i]>1) return i;
        }

        return -1;
        
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cs
public class Solution
{
    public int FindDuplicate(int[] nums)
    {
        Dictionary<int, int> m = new Dictionary<int, int>();

        foreach (int num in nums)
        {
            if (m.ContainsKey(num))
                m[num]++;
            else
                m[num] = 1;
        }

        foreach (var ele in m)
        {
            if (ele.Value > 1)
                return ele.Key;
        }

        return -1; // no duplicate found
    }
}
